perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_lifescience_hostel_persistent_shadow.java

```diff
@@ -0,0 +1,137 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.exceptions.*;
+import cz.metacentrum.perun.core.blImpl.ModulesUtilsBlImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.ExtSource;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.UserExtSource;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+
+/**
+ * Class for checking logins uniqueness in the namespace and filling lifescience-hostel-persistent id.
+ * It is only storage! Use module login lifescience-hostel_persistent for access the value.
+ *
+ * @author Peter Jancus <p.jancus1996@gmail.com>
+ */
+public class urn_perun_user_attribute_def_def_login_namespace_lifescience_hostel_persistent_shadow extends urn_perun_user_attribute_def_def_login_namespace {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_def_login_namespace_bbmri_test_persistent_shadow.class);
+	private final static String extSourceNameLSH = "https://login.bbmri-eric.eu/idp/";
+
+	/**
+	 * Filling implemented for login:namespace:lifescience-hostel-persistent attribute
+	 *
+	 * Resulting format/rules:
```

Comment:
Implementation also crops resulting login to 20 chars, since its limit of MS Active Directory. But there is no indication, if it will be used/stored in AD, also javadoc doesn't specify it. So please make sure, what it supposed to generate and how it will be used. Then change the code or javadoc respectively.

Also we recently find out a problem, when user name is filled with eg. Chinese characters. Then even if name is not empty, it's not used to generate the login and this logic fails. We should make sure, that on empty result we get something for backup.